code/controllers/subsystem/mapping.dm

```diff
@@ -77,6 +91,56 @@ SUBSYSTEM_DEF(mapping)
 
 	return ..()
 
+// Do not confuse with seedRuins()
+/datum/controller/subsystem/mapping/proc/handleRuins()
+	// load in extra levels of space ruins
+	var/load_zlevels_timer = start_watch()
+	log_startup_progress("Creating random space levels...")
+	var/num_extra_space = rand(config.extra_space_ruin_levels_min, config.extra_space_ruin_levels_max)
+	for(var/i = 1, i <= num_extra_space, i++)
+		GLOB.space_manager.add_new_zlevel("Ruin Area #[i]", linkage = CROSSLINKED, traits = list(REACHABLE, SPAWN_RUINS))
+	log_startup_progress("Loaded random space levels in [stop_watch(load_zlevels_timer)]s.")
+
+	// Now spawn ruins, random budget between 20 and 30 for all zlevels combined.
+	// While this may seem like a high number, the amount of ruin Z levels can be anywhere between 3 and 7.
+	// Note that this budget is not split evenly accross all zlevels
+	log_startup_progress("Seeding ruins...")
+	var/seed_ruins_timer = start_watch()
+	seedRuins(levels_by_trait(SPAWN_RUINS), rand(20, 30), /area/space, GLOB.space_ruins_templates)
+	log_startup_progress("Successfully seeded ruins in [stop_watch(seed_ruins_timer)]s.")
+
+// Loads in the station
+/datum/controller/subsystem/mapping/proc/loadStation()
+	ASSERT(map_datum.map_path)
+	if(!fexists(map_datum.map_path))
+		// Make a VERY OBVIOUS error
+		to_chat(world, "<span class='narsie'>ERROR: The path specified for the map to load is invalid. No station has been loaded!</span>")
+		return
```

Comment:
This will never happen in a prod scenario and should only appear to people on local test boxed, who will see it if they are launching with the debugger (F5), which force logs them in

Reply:
I know that some people prefer running it through dream daemon which is why I mentioned, but it'll be fine in VSCode yeah.